controllers/iamrole_controller.go

```diff
@@ -140,16 +142,30 @@ func (r *IamroleReconciler) HandleReconcile(ctx context.Context, req ctrl.Reques
 		ns = *ns2
 	}
 
+	// Figure out the name of the IAM Role itself that will be created based on the Iamrole resource name, namespace,
+	// and the configuration settings.
 	roleName, err := utils.GenerateRoleName(ctx, iamRole, *config.Props, &ns)
-	log.V(1).Info("roleName constructed successfully", "roleName", roleName)
-
 	if err != nil {
 		r.Recorder.Event(iamRole, v1.EventTypeWarning, string(iammanagerv1alpha1.Error), "Unable to construct iam role name to error "+err.Error())
 		// It is not clear to me that we want to requeue here - as this is a fairly permanent
 		// error. Is there a better pattern here?
 		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
 	}
 
+	// Determine whether or not this Iamrole is supposed to be connected to a ServiceAccount or not. This is
+	// used during the 'Ready' and 'default' status case handlers below to ensure the connection between the
+	// ServiceAccount and IAM Role (if desired).
+	shouldManageSa, saName := utils.ParseIRSAAnnotation(ctx, iamRole)
+
+	// Get a RBAC object we can use to modify the ServiceAccount if we need to.
+	rbac, err := k8s.NewK8sClient()
```

Comment:
We probably shouldn't do this since there are users who doesn't use IRSA and getting the client at the front doesn't like a good idea. We should get the client only if it is needed- For example: If shouldManageSa -> get the reusable client so you can use it to get the service account for the comparison and update the service account at the end if needed